apps/server/src/modules/system/service/system.service.ts

```diff
@@ -1,27 +1,126 @@
 import { Injectable } from '@nestjs/common';
+import { EntityId, System, SystemTypeEnum } from '@shared/domain';
 import { SystemRepo } from '@shared/repo';
 import { SystemMapper } from '@src/modules/system/mapper/system.mapper';
 import { SystemDto } from '@src/modules/system/service/dto/system.dto';
-import { EntityId, System } from '@shared/domain';
 
 @Injectable()
 export class SystemService {
 	constructor(private readonly systemRepo: SystemRepo) {}
 
-	async find(type: string | undefined, onlyOauth = false): Promise<SystemDto[]> {
+	async find(type: string | undefined): Promise<SystemDto[]> {
 		let systemEntities: System[];
-		if (type === '' && !onlyOauth) {
+		if (!type) {
 			systemEntities = await this.systemRepo.findAll();
 		} else {
-			systemEntities = await this.systemRepo.findByFilter(type, onlyOauth);
+			systemEntities = await this.systemRepo.findByFilter(type, false);
 		}
-		const systemDtos: SystemDto[] = SystemMapper.mapFromEntitiesToDtos(systemEntities);
-		return systemDtos;
+		return SystemMapper.mapFromEntitiesToDtos(systemEntities);
 	}
 
 	async findById(id: EntityId): Promise<SystemDto> {
-		const entity: System = await this.systemRepo.findById(id);
-		const systemDto: SystemDto = SystemMapper.mapFromEntityToDto(entity);
-		return systemDto;
+		const entity = await this.systemRepo.findById(id);
+		return SystemMapper.mapFromEntityToDto(entity);
+	}
+
+	/*
+	This returns systems that provide authentication via OAuth
+	It merges systems that dBildungscloud communicates with directly
+	and systems that are queried via keycloak brokering.
+	*/
+	async findOAuth(): Promise<SystemDto[]> {
+		const systemEntities = await this.findDirectOauthSystems();
+		const generatedOAuthsystems = await this.findOauthViaKeycloakSystems();
+
+		return [...systemEntities, ...generatedOAuthsystems];
+	}
+
+	async findOidc(): Promise<SystemDto[]> {
+		return this.findOIDCSystems();
```

Comment:
We could make `findOIDCSystems()` public instead

Reply:
Done